### PR TITLE
Resolve "New Block" form still show "Store View" field in "Single Store Mode" issue24387

### LINKED ISSUE #24387: "New Block" form still show "Store View" field in "Single Store Mode"
### Preconditions (*)

1. Magento 2.3-develop
2. Enable "Single Store Mode"

image

### Steps to reproduce (*)

1. Go to backend
2. Content->Blocks
3. Add New Block
4. See:

image

### Expected result (*)

1. No Store View field

### Actual result (*)

image

### PR DESCRIPTION
**Pull request by edenduong**
### Description (*)

1. Resolve magento/magento2#24387: "New Block" form still show "Store View" field in "Single Store Mode" 

Solution: Add the class to handle the case `$this->storeManager->isSingleStoreMode()
`
### Fixed Issues (if relevant)

1. magento/magento2#24387: "New Block" form still show "Store View" field in "Single Store Mode" 
### Manual testing scenarios (*)

1. Magento 2.3-develop
2. Enable "Single Store Mode"
image
3. Go to backend
4. Content->Blocks
5. Add New Block
=> no Store View field in the form

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Cms/Ui/Component/Form/Field/StoreView.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Cms\Ui\Component\Form\Field;
+
+use Magento\Framework\View\Element\UiComponent\ContextInterface;
+use Magento\Framework\View\Element\UiComponentFactory;
+use Magento\Framework\View\Element\UiComponentInterface;
```
Comment (swnsma) [inline app/code/Magento/Cms/Ui/Component/Form/Field/StoreView.php]: In current implementation, this `use` is redundant.

File: app/code/Magento/Cms/Ui/Component/Form/Field/StoreView.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+use Magento\Store\Model\StoreManagerInterface as StoreManager;
+use Magento\Ui\Component\Form\Field;
+
+/**
+ * Check to disable store view field
+ *
+ * Class \Magento\Cms\Ui\Component\Form\Field\StoreView
+ */
+class StoreView extends Field
```
Comment (swnsma) [inline app/code/Magento/Cms/Ui/Component/Form/Field/StoreView.php]: I am not sure if this class should be placed right here.
Could we move this to `Magento\Store\Ui\Component\Form\Field\StoreView`?

**Review by swnsma — CHANGES_REQUESTED** [review]
Please check my comments.

**Comment by edenduong** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @edenduong. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @edenduong, here is your [new Magento instance](https://pr-24397.instances.magento-community.engineering).
Admin access: [https://pr-24397.instances.magento-community.engineering/admin](https://pr-24397.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

**Comment by edenduong** [comment]
> Please check my comments.

I have changed the source code. Please check it again. Thanks ! @swnsma

**Review by swnsma — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @swnsma, thank you for the review.
[ENGCOM-5762](https://jira.corp.magento.com/browse/ENGCOM-5762) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Pull request merged**